src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala

```diff
@@ -220,14 +89,15 @@ object BasicOpcDiff extends OpcDiffer {
       differ: (Array[Byte], Seq[DataEntry]) => Either[ValidationError, Seq[DataEntry]])
     extends Val(basicType) { def *(n: Int): Int = n * basicType }
 
-    val Add         = basicTypeVal(1, 4, (b, d) => add(d(b(1)), d(b(2)), d, b(3)))
-    val Minus       = basicTypeVal(2, 4, (b, d) => minus(d(b(1)), d(b(2)), d, b(3)))
-    val Multiply    = basicTypeVal(3, 4, (b, d) => multiply(d(b(1)), d(b(2)), d, b(3)))
-    val Divide      = basicTypeVal(4, 4, (b, d) => divide(d(b(1)), d(b(2)), d, b(3)))
-    val Minimum     = basicTypeVal(5, 4, (b, d) => minimum(d(b(1)), d(b(2)), d, b(3)))
-    val Maximum     = basicTypeVal(6, 4, (b, d) => maximum(d(b(1)), d(b(2)), d, b(3)))
+    val Add         = basicTypeVal(1, 4, (b, d) => operation(d(b(1)), d(b(2)), d, b(3), add))
+    val Minus       = basicTypeVal(2, 4, (b, d) => operation(d(b(1)), d(b(2)), d, b(3), minus))
+    val Multiply    = basicTypeVal(3, 4, (b, d) => operation(d(b(1)), d(b(2)), d, b(3), multiply))
+    val Divide      = basicTypeVal(4, 4, (b, d) => operation(d(b(1)), d(b(2)), d, b(3), divide))
+    val Minimum     = basicTypeVal(5, 4, (b, d) => operation(d(b(1)), d(b(2)), d, b(3), minimum))
+    val Maximum     = basicTypeVal(6, 4, (b, d) => operation(d(b(1)), d(b(2)), d, b(3), maximum))
     val Concat      = basicTypeVal(7, 4, (b, d) => concat(d(b(1)), d(b(2)), d, b(3)))
     val ConstantGet = basicTypeVal(8, 2, (b, d) => constantGet(b.slice(1, b.length-1), d, b(b.length-1)))
+    val SqrtBigInt  = basicTypeVal(9, 3, (b, d) => sqrt(d(b(1)), d, b(2)))
```

Comment:
Nice simplification. 
1. We need extra OPC which transform the dataEntry between `Amount`, `Int32`, `Timestamp` and `BigInteger `
2. and, or, xor her only for Boolean type
3. comparison OPC needs to support BigInteger in `CompareOpcDiff`. with extra `>`, `<`, `<=`

But we may merge this PR first and add these new features/functions in next PR.

Reply:
ok, I'll do it in other pr